library/src-bootstrapped/scala/quoted/Liftable.scala

```diff
@@ -305,4 +306,27 @@ object Liftable {
       '{ BigDecimal(${Expr(x.toString)}) }
   }
 
+  inline given productLiftable[T <: Product: Type](given
```

Comment:
To avoid duplication of most of this logic you can split it into 
```scala
inline given [T <: Product: Type](given
    m : Mirror.ProductOf[T],
    em: Expr[Mirror.ProductOf[T]]): Liftable[T] =
      productLiftable(given m, em)(summonAll[Tuple.Map[m.MirroredElemTypes, Liftable]])

private def productLiftable[T <: Product: Type](given
    m : Mirror.ProductOf[T],
    em: Expr[Mirror.ProductOf[T]])(liftables: List[Any]): Liftable[T] = new Liftable[T] {
  def toExpr(x: T) =
    val genRepr = Tuple.fromProductTyped(x)
    val elemsWithLiftables = liftables.zip(genRepr.asInstanceOf[Product].productIterator.toList)
    val tupleOfExprs = elemsWithLiftables.map {
      case (l: Liftable[a], x) => l.toExpr(x.asInstanceOf[a])
    }
    val exprOfTuple = Expr.ofTuple(tupleOfExprs)
    '{$em.fromProduct($exprOfTuple.asInstanceOf[Product])}
}
```

Reply:
To avoid inlining too much